packages/toc/src/generators/notebook/utils.ts

```diff
@@ -0,0 +1,49 @@
+import { MarkdownCell } from '@jupyterlab/cells';
+import { Cell } from '@jupyterlab/cells';
+
+/**
+ * Determines the collapsed state of a header in the ToC. If collapsibleNotebooks setting
+ * is set to true, toc state is also updated to match the state of the Notebook header
+ *
+ * @param cell - cell containing the header in the notebook
+ * @param collapsibleNotebooks - setting indicating if header cell collapse state and ToC collapse state should match
+ */
+
+export function headerCollapsedState(
+  cell?: Cell,
+  collapsibleNotebooks?: boolean
+): boolean {
+  let collapsed;
+  let tocCollapsed;
+  if (cell!.model.metadata.has('toc-hr-collapsed')) {
```

Comment:
It could be in a follow-up PR, but I think we might want to move to using the same metadata as the notebook cells are using to indicate collapse here: https://github.com/jupyterlab/jupyterlab/blob/master/packages/cells/src/widget.ts#L1435

Reply:
Thanks for pointing that out! With that in mind, I think instead of determining collapse state from reading metadata and cell.headingCollapsed as I do in this function, I should probably toggle what metadata the ToC is checking on the cell based on what the `syncCollapseState`. So if `syncCollapseState` is is false, I would still use `toc-hr-collapsed`, but if it were True, I would use `MARKDOWN_HEADING_COLLAPSED` instead. Then I could get rid of these util functions altogether and keep the code closer to the original state. Is that how you would handle it?

Reply:
Yes that seems good!